tests/util/localregistry/localregistry.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package localregistry
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"istio.io/istio/tests/util"
+)
+
+const (
+	// Default values for test env setup
+	LocalRegistryFile      = "tests/util/localregistry/localregistry.yaml"
+	LocalRegistryNamespace = "kube-system"
+	LocalRegistryPort      = 5000
+)
+
+// SetupLocalRegistry sets up localregistry and returns if any error was found while doing that.
+func SetupLocalRegistry(remoteRegistryPort uint16, kubeconfig string) error {
+	goPath := os.Getenv("GOPATH")
+	if len(goPath) == 0 {
+		return fmt.Errorf("GOPATH not set.")
+	}
+
+	localRegistryFilePath := path.Join(util.GetResourcePath(LocalRegistryFile))
+	if err := util.KubeApply(LocalRegistryNamespace, localRegistryFilePath, kubeconfig); err != nil {
+		return fmt.Errorf("Kubectl apply %s failed", LocalRegistryFile)
+	}
+
+	if err := checkLocalRegistryRunning(); err != nil {
```

Comment:
Let's be consistent with the new framework for things like this.  See https://github.com/istio/istio/blob/master/pkg/test/framework/components/policybackend/policybackend.go#L189

In general, consider switching to kube client rather than just invoking kubectl. It will ultimately require less work when doing complex operations.

Reply:
+1 in favor of kube client

Reply:
But I think we can switch to kube client in follow-ups - I am happy to send another PR when this PR get merged.

WDYT @nmittler

Reply:
@m1093782566 that's fair.  I've created https://github.com/istio/istio/issues/8877 to track.